sa/sa.go

```diff
@@ -32,7 +33,10 @@ import (
 	sapb "github.com/letsencrypt/boulder/sa/proto"
 )
 
-var errIncompleteRequest = errors.New("incomplete gRPC request message")
+var (
+	errIncompleteRequest     = errors.New("incomplete gRPC request message")
+	validIncidentTableRegexp = regexp.MustCompile(`(?m)^incident_[a-zA-Z_]{1,91}$`)
```

Comment:
`(?m)` turns on multi-line mode, which you don't need here since newlines are not accepted as part of your pattern. https://github.com/google/re2/wiki/Syntax

The regex should accept numerals as well (sorry I forgot to include that in my proposed syntax).

The 100 char limit I proposed was pretty arbitrary - rather than subtracting the length of `incident_`, want to just make the requirement `{1,100}`?

Reply:
> (?m) turns on multi-line mode, which you don't need here since newlines are not accepted as part of your pattern. https://github.com/google/re2/wiki/Syntax

Ah, yeah I realized this and forgot to remove it, good catch.

> The regex should accept numerals as well (sorry I forgot to include that in my proposed syntax).
>
> The 100 char limit I proposed was pretty arbitrary - rather than subtracting the length of incident_, want to just make the requirement {1,100}?

👍🏻

Reply:
😅 well, it appears that my first attempt was actually correct. Disabling multi-line mode means that this regex will still match strings in excess of 100 characters. Immediately after removing this line I began receiving MySQL `ER_WRONG_TABLE_NAME` errors for  'loooooooooooo...' in the unit tests.

Reply:
> Disabling multi-line mode means that this regex will still match strings in excess of 100 characters.

This seems really surprising. Can you provide a minimal repro on play.golang.org?

Reply:
https://go.dev/play/p/Q06uqK5d02I

Reply:
```
	valid      = regexp.MustCompile(`^incident_[0-9a-zA-Z_]{1,20}$`)
	validMulti = regexp.MustCompile(`(?m)^incident_[0-9a-zA-Z_]{1,20}$`)
```

Reply:
Ah, I could have sworn that multi-line had to be enabled to use the line terminator symbols! Thanks.